server/routes/experiments/biomed_nl/utils.py

```diff
@@ -307,6 +309,18 @@
 MAX_NUM_DCID_PER_V2NODE_REQUEST = 100
 
 
+class GraphEntity(BaseModel):
+  name: str
+  types: list[str]
+  dcid: str
+
+  def __eq__(self, other):
+    if isinstance(other, GraphEntity):
+      return (self.name == other.name and self.dcid == other.dcid and
+              Counter(self.types) == Counter(other.types))
```

Comment:
could there be more than 1 of each type? could just use set(self.types) == set(other.types) if each type should only appear once?